### ISSUE TITLE
Tooltip not getting overlapping values if XAxis is set to scale time

### ISSUE THREAD

**Issue by bandinopla** [recharts/recharts#3387]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/recharts-issue-template-forked-jz8yup?file=/src/index.js)

### Steps to reproduce
rendering 2 Line components with 2 distinct data sets, using the X axis as time, if both overlap, and you set a tooltip, the tooltip doesn't display the combined data on that point like it does when you set the XAxis to "category" with allowDuplicates = false.

### What is expected?
if i put the mouse on an X Point, i will expect the tooltip to be provided with only the data on that X Point. 

### What is actually happening?
i get the data from one line in the tooltip then if i move the mouse a little i see the other overlapping point data.

| Environment | Info |
|---|---|
| Recharts | v2.1.2 |
| React | ^17.0.2 |
| System | Windows |
| Browser | Brave v1.48.167 |

**Comment by ckifer** [comment]
Can you post a screenshot of what you mean? Its not clear based on the issue description.

Are you talking about how the example is including `uv` from `data` in the tooltip as opposed to nothing?

![image](https://user-images.githubusercontent.com/25180830/220430610-8f49f8b6-a10f-4f98-b22e-8785cc586968.png)

**Comment by bandinopla** [comment]
@ckifer yes, if i hover the cursor over an X Point that has only 1 point in it, i would expect the "payload" from the tooltip to only contains the points in that position. This works fine if you use an XAxis of type category, but when using a numeric X Axis for some reason the tooltip only gets one value at a time. 

One quick fix im doing right now is, when i hover, in the custom tooltip, i get the label of the XAxis which in my case is a timestamp, then i use that to manually relocate all points in all series that have that same X value and build the tooltip from that data. 

I was expecting to get this automatically vía the payload property, but i always get all series even if they dont contain any point in the X position and the payload of those irrelevant series contain also a payload pointing to a point that is on a diferent X...

**Comment by dkstrong** [comment]
I'm having the same issue, I believe the core problem is that its trying to find the other payload items by their index. however since each data set has  different number of items you can't use the index to match/compare them. I think it needs to be looking at the dataKey for the x axis to figure out the matching payloads.

**Comment by dusan-w3lab** [comment]
Any updates about this bug? @ckifer @bandinopla

**Comment by ckifer** [comment]
@dusan-w3lab being worked on in https://github.com/recharts/recharts/pull/3621

**Comment by ana1sokol** [comment]
Is this issue going to be fixed?
